src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -173,17 +183,23 @@ private void removeDeadDataNodes() {
     return allTasksIds.values();
   }
 
-  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType) {
+  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType, String taskName) {
     switch (taskType) {
       case HDFSConstants.NAME_NODE_ID :
         HashMap<String, String> nameNodes = getNameNodes();
         nameNodes.put(hostname, taskId.getValue());
         setNameNodes(nameNodes);
+        HashMap<String, String> nameNodeTaskNames = getNameNodeTaskNames();
+        nameNodeTaskNames.put(taskId.getValue(), taskName);
```

Comment:
We need to remove by taskId because that's what we know when the task is removed so this way is more convenient. Even if we reversed this map, we would have to call keys instead of values.